app/src/main/java/com/azuresamples/msalandroidapp/MainActivity.java

```diff
@@ -89,44 +142,73 @@ public void onAccountsLoaded(final List<IAccount> accounts) {
      * Callback will call Graph api w/ access token & update UI
      */
     private void onCallGraphClicked() {
-        sampleApp.acquireToken(getActivity(), SCOPES, getAuthInteractiveCallback());
+        //sampleApp.acquireToken(getActivity(), SCOPES, getAuthInteractiveCallback());
+        sampleApp.signIn(getActivity(), SCOPES, new AuthenticationCallback() {
+            @Override
+            public void onSuccess(IAuthenticationResult authenticationResult) {
+                getSilentToken();
+            }
+
+            @Override
+            public void onError(MsalException exception) {
+
+            }
+
+            @Override
+            public void onCancel() {
+
+            }
+        });
     }
 
     /* Clears an account's tokens from the cache.
      * Logically similar to "sign out" but only signs out of this app.
      * User will get interactive SSO if trying to sign back-in.
      */
     private void onSignOutClicked() {
+
+
         /* Attempt to get a user and acquireTokenSilent
          * If this fails we do an interactive request
          */
-        sampleApp.getAccounts(new PublicClientApplication.AccountsLoadedCallback() {
+        sampleApp.getCurrentAccountAsync(new ISingleAccountPublicClientApplication.CurrentAccountCallback() {
             @Override
-            public void onAccountsLoaded(final List<IAccount> accounts) {
-
-                if (accounts.isEmpty()) {
-                    /* No accounts to remove */
-
-                } else {
-                    for (final IAccount account : accounts) {
-                        sampleApp.removeAccount(
-                                account,
-                                new PublicClientApplication.AccountsRemovedCallback() {
-                            @Override
-                            public void onAccountsRemoved(Boolean isSuccess) {
-                                if (isSuccess) {
-                                    /* successfully removed account */
-                                } else {
-                                    /* failed to remove account */
-                                }
-                            }
-                        });
-                    }
+            public void onAccountLoaded(@Nullable IAccount activeAccount) {
+                if(activeAccount != null){
+                    signOut();
                 }
+            }
 
+            @Override
+            public void onAccountChanged(@Nullable IAccount priorAccount, @Nullable IAccount currentAccount) {
+                if(currentAccount != null){
+                    signOut();
+                }
+            }
+
+            @Override
+            public void onError(@NonNull Exception exception) {
+
```

Comment:
We should show a `Toast` or dialog if this error gets hit - otherwise, users of the samples might be confused with a silent failure